### PR TITLE
cli: Cloud traces support via `dagger login`

### PR DESCRIPTION
**Pull request by aluzzardi**

### CODE REVIEW

**Comment by aluzzardi** [comment]
@vito @sipsma Just realized this only supports traces, not magicache. Do we care?

**Comment by jedevc** [comment]
@aluzzardi I think I'd *expect* `dagger login` to do magicache, but also... not sure how much effort it's worth investing in it, since magicache is fairly tied to the engine lifetime, not the client lifetime. Since we're looking at storage drivers as the future of this, I don't think it's necessary - we can just note this in the docs.

**Comment by vito** [comment]
Pushed a commit to fix `os.Stderr` and `os.Exit` usage which was breaking the TUI, but I'm also seeing it not update my `credentials.json` file, looking into it

**Review by vito — APPROVED** [review]
Confirmed it works after my last few fixes :+1: - tested against the preview env.

**Review by vito — APPROVED** [review]

File: cmd/dagger/cloud.go [hunk 1]
```diff
@@ -13,7 +13,6 @@ import (
 func init() {
 	cloud := &CloudCLI{}
 
-	rootCmd.PersistentFlags().StringVar(&cloud.API, "api", "https://api.dagger.cloud", "Dagger Cloud API URL")
 	rootCmd.PersistentFlags().MarkHidden("api")
```
[the remark this answers is not in the document]
Comment (aluzzardi) [reply cmd/dagger/cloud.go:16]: Thank for you the review! LGTM'd the follow ups

**Pull request merged**